packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -871,6 +874,97 @@ impl<'m> RunnerImpl<'m> {
         Ok(())
     }
 
+    /// TODO
+    fn handle_task_msg(
+        &mut self,
+        mv8: &'m MiniV8,
+        sim_id: SimulationShortId,
+        msg: RunnerTaskMsg,
+        outbound_sender: &UnboundedSender<OutboundFromRunnerMsg>,
+    ) -> Result<()> {
+        tracing::debug!("Starting state interim sync before running task");
+        // TODO: Move JS part of sync into `run_task` function in JS for better performance.
+        self.state_interim_sync(mv8, sim_id, &msg.shared_store)?;
+
+        tracing::debug!("Setting up run_task function call");
+
+        let (payload, wrapper) = msg
+            .payload
+            .extract_inner_msg_with_wrapper()
+            .map_err(|err| {
+                Error::from(format!("Failed to extract the inner task message: {err}"))
+            })?;
+        let payload_str = mv8::Value::String(mv8.create_string(&serde_json::to_string(&payload)?));
+
+        let context = msg.shared_store.context().clone();
```

Comment:
Can we move out `context` from `shared_store` i.e. destructuring `shared_store` here?

Reply:
`context` is private, if you go and look at the class you might see part of the reasoning behind it